### ISSUE TITLE
Document data generation process

### ISSUE THREAD

**Issue by chrispahm** [carderne/zarr-gl#2]
Hi @carderne,

Thanks a lot for this nice library!
I know you just released an early version of the library but would it be possible to get more info on how to create a compatible .zarr pyramid?
I've tried guessing the relevant steps but couldn't get it to work so far. Here's what I tried (in Python):

```py
# df_4326 is a GeoPandas dataframe
out_grid = make_geocube(df_4326, measurements=["SE600.2020"], resolution=(-1, 1))
out_grid = out_grid.rio.write_crs('EPSG:4326')
pyramid = pyramid_reproject(out_grid, levels=6)
pyramid.to_zarr('FADN_Spatial.zarr')
```

The webpage is a bare minimum HTML page (no react) with the following simple script:

```html
<script type="module">
      import zarrgl from 'https://cdn.jsdelivr.net/npm/zarr-gl@0.1.4/+esm'
      import {makeColormap} from 'https://cdn.jsdelivr.net/npm/@carbonplan/colormaps@4.0.0/+esm'

      const map = new maplibregl.Map({
        container: "map",
        style: "https://demotiles.maplibre.org/style.json",
        center: [
          7.221275, 50.326111
        ],
        zoom: 2
      });

      const colormap = makeColormap('warm', {
        count: 7,
        mode: 'light',
        format: 'hex'
      })

      const breaks = {
        "SE600": [
          -5523.0,
          120.0,
          153.0,
          184.0,
          223.0,
          281.0,
          356.0,
          458.0,
          22970.0
        ]
      }

      map.on("load", () => {
        const layer = new zarrgl.ZarrLayer({
          id: "myZarrLayer",
          source: "https://url/to/File.zarr",
          variable: "SE600.2020",
          colormap,
          clim: [
            breaks["SE600"][0],
            breaks["SE600"][breaks["SE600"].length - 1]
          ],
          opacity: 0.8,
          map
        });
        map.addLayer(layer);
      })
    </script>
```

On the page I just get a gray overlay on top of the map, and a console warning stating `WebGL: INVALID_OPERATION: texImage2D: ArrayBufferView not big enough for request`

<img width="1143" alt="image" src="https://github.com/user-attachments/assets/5f919e7f-85b8-414d-a93f-96fa3f64b1ec">

Is there any obvious issue with this procedure or the way I'm generating the zarr pyramid? I also attached the raster GeoTif and the Zarr as zip files. Thanks a lot for you help!

[FADN_Spatial.tif.zip](https://github.com/user-attachments/files/17478337/FADN_Spatial.tif.zip)
[FADN_Spatial.zarr.zip](https://github.com/user-attachments/files/17478333/FADN_Spatial.zarr.zip)

**Comment by carderne** [comment]
Yes working on it now! It's not at all self-explanatory and requires quite a bit of pre-processing, but I'm hoping there'll be a way around that ultimately.

Will ping you here once I've got something up, hopefully tomorrow.

**Comment by carderne** [comment]
Hey @chrispahm I just added this:

https://github.com/carderne/zarr-gl/tree/main/example

Let me know if that helps and works for you! Happy to elaborate if useful but tbh I just started looking into this a week ago and not sure about the rough edges yet...

Unfortunately with the change to zarrita and microbundle the jsDelivr script isn't working so I've pinned it to 0.1.5. Will try figure that out or roll it back, JS bundling a bit of a mystery.

**Comment by carderne** [comment]
Oh a few important caveats that I should probably add to the README:
- The tile size is currently hardcoded to 128. Just a few lines to pull the from zattrs just need to get to it.
- Ditto the nodata value
- It's currently all built on the assumption of Float32 data, not sure what exatly I need to do to get it to accept others.

**Comment by carderne** [comment]
jsdelivr-style import should now work with any version >= 0.1.7.

**Comment by chrispahm** [comment]
Awesome @carderne, thanks a lot for your help! After fiddling around for a bit I found out that my colormap array consisting of hex strings was the culprit here! Thanks for providing that basic example that gives an idea of what the colormap array should look like!

**Comment by chrispahm** [comment]
Hey sorry for the late reply! I was also just fiddling around with the library because I like the general possibility of using the zarr format for raster data like I'd use PMTiles for vector data.  

The only things I was missing for now was the possibility to specify "nodata" values (and a color), as well as more fine grained control over the colormap. Both might already be possible right now, but at least I couldn't see it right away when inspecting the `ZarrLayerProps` interface (https://github.com/carderne/zarr-gl/blob/main/src/index.ts). Maybe a very short API section in the docs would be helpful here? 

With the colormap options is was thinking about the ability to use anything different compared to linear scales, but again just a thought after briefly fiddling with the data.
